src/backend/query_builder.rs

```diff
@@ -1215,13 +1214,22 @@ pub trait QueryBuilder: QuotedBuilder + EscapeBuilder + TableRefBuilder {
     /// Translate part of a condition to part of a "WHERE" clause.
     fn prepare_condition_where(&self, condition: &Condition, sql: &mut dyn SqlWriter) {
         if condition.negate {
-            write!(sql, "NOT (").unwrap();
+            write!(sql, "NOT ").unwrap()
         }
-        let mut is_first = true;
-        for cond in &condition.conditions {
-            if is_first {
-                is_first = false;
-            } else {
+
+        if condition.is_empty() {
+            match condition.condition_type {
+                ConditionType::All => self.prepare_constant(&true.into(), sql),
+                ConditionType::Any => self.prepare_constant(&false.into(), sql),
```

Comment:
I think we flipped the logic here (or this is a deep topic?)
`ANY of ()`: anything from an empty set, so `TRUE`?
`ALL of ()`: all of the set, which is empty, so much be `FALSE`?